how-to-handle-history-mode-files.md

```diff
@@ -0,0 +1,220 @@
+#### What is History Mode
+
+History mode allows us to capture every version of each record processed by the fivetran connectors.
+In order to keep all versions of the record, we have introduced three new system columns for tables with history mode enabled.
+
+
+Column | Type | Description
+--- | --- | ---
+_fivetran_active | Boolean | TRUE if it is the currently active record. FALSE if it is a historical version of the record. Only one version of the record can be TRUE.
+_fivatran_start | TimeStamp | The time when the record was first created or modified in the source.
+_fivetran_end | TimeStamp | The value for this column depends on whether the record is active. If the record is not active(`_fivetran_active`=FALSE), then `_fivetran_end` value will be `_fivetran_start` of the next version of the record minus 1 millisecond. If the record is deleted, then the value will be the same as the timestamp of delete operation. If the record is active(`_fivetran_active`=TRUE), then `_fivetran_end` is the max allowed value that we can set for a TIMESTAMP column.
+
+
+#### Points to remember in history mode
+
+- In WriterBatchRequest we pass a new optional field HistoryMode which indicates connector is in history mode or not. In this HistoryMode field, we pass `deleted_column` column name which we need to modify only if it is present in destination.   
+- If the existing table is not empty then in the batch file we also send a boolean column `_fivetran_earliest`. Suppose in an `upsert` we got multiple versions of the same record in a flush, then we set the `_fivetran_earliest` to `TRUE` for the record which have the earliest `_fivetran_start` and rest of the versions will have `_fivetran_earliest` as FALSE.
+- For each Replace, Update and Delete batch files, DELETE the existing records from destination table if `_fivetran_start` of destination table is greater than or equal to  `_fivetran_start` of batch file(Refer Replace example 1 and example 2).
+
+Note: The `_fivetran_earliest` column shouldn't be added in the destination table. It is introduced to easily identify the earliest record and can be used to optimize data loads query.
+Below is an example of `replace_file`
+
+Id(PK) | COL1    | _fivetran_start(PK) | _fivetran_end | _fivetran_active | _fivetran_earliest
+---|---------|---------------------| --- |------------------| --- 
+1 | abc     | T1                  | T2-1  | FALSE            | TRUE
+2 | xyz     | T1                  | TMAX | TRUE             | TRUE
+1 | pqr     | T2                  | T3-1 | FALSE            | FALSE
+1 | def | T3                  | TMAX                | TRUE             | FALSE
+
+#### How to Handle Replaces, Updates and Deletes
+
+##### Replace
+
+###### Example 1:
+
+When `_fivetran_start` of destination table is less than  `_fivetran_start` of batch file.
```

Comment:
```suggestion
When the `_fivetran_start` column value of destination table is less than  `_fivetran_start` of batch file.
```
Incomplete sentence.

Reply:
I am trying to add 2 examples:
 Example 1: When  the `_fivetran_start` column value in the destination table is less than  `_fivetran_start` of the batch file.
Example2: When the `_fivetran_start` column value in the destination table is greater than or equal to the `_fivetran_start` in the batch file.